ql/experimental/volatility/sviinterpolation.hpp

```diff
@@ -33,16 +33,16 @@ namespace QuantLib {
 namespace detail {
 
 inline void checkSviParameters(const Real a, const Real b, const Real sigma,
-                               const Real rho, const Real m) {
+                               const Real rho, const Real m, const Time tte) {
     QL_REQUIRE(b >= 0.0, "b (" << b << ") must be non negative");
     QL_REQUIRE(std::fabs(rho) < 1.0, "rho (" << rho << ") must be in (-1,1)");
     QL_REQUIRE(sigma > 0.0, "sigma (" << sigma << ") must be positive");
     QL_REQUIRE(a + b * sigma * std::sqrt(1.0 - rho * rho) >= 0.0,
                "a + b sigma sqrt(1-rho^2) (a=" << a << ", b=" << b << ", sigma="
                                                << sigma << ", rho=" << rho
                                                << ") must be non negative");
-    QL_REQUIRE(b * (1.0 + std::fabs(rho)) < 4.0,
-               "b(1+|rho|) must be less than 4");
+    QL_REQUIRE(b * (1.0 + std::fabs(rho)) <= 4.0 / tte,
```

Comment:
We should probably check for `tte != 0` before dividing.  And if tte is 0, the check would pass, I guess?

Reply:
Possibly so, yes. I skipped it here since I read the condition [here](https://github.com/lballabio/QuantLib/blob/14c2ecbdd2e712feabbff8ee46b0b1efa2dabb77/ql/termstructures/volatility/smilesection.cpp#L67) as strictly greater than 0, but I realize now that it allows for equality too. I'm not sure what would be the correct way to do it now. In the current implementation, [this line](https://github.com/lballabio/QuantLib/blob/14c2ecbdd2e712feabbff8ee46b0b1efa2dabb77/ql/experimental/volatility/svismilesection.cpp#L52) in `volatilityImpl()` also seems like it would fail now if `t == 0`?

Reply:
Hmm, you're right.  We can probably be stricter in `SviSmileSection` than in its base class and require t > 0.

As for the check, that's independent of the class.  I guess if tte = 0, the check passes as the right side goes to infinity.